server/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -489,4 +508,28 @@ public void writeTo(StreamOutput out) throws IOException {
             failure.writeTo(out);
         }
     }
+
+    public void writeThin(StreamOutput out) throws IOException {
```

Comment:
This seems to leave `BulkItemResponse#writeTo()` as dead code that mostly duplicates this. Does that mean that we don't need `BulkItemResponse implements Writeable`? If so, can you remove that and rename these methods more appropriately?

Reply:
We're still using it in `org.elasticsearch.action.bulk.BulkItemRequest#writeTo` (though now that I looked over that, we might not have to ever write the `ShardId` for that one as well logically because it's always part of a `BulkShardRequest` which never has `null` for the `ShardId`).
To me it seems easier to leave it `Writable` for now so we have the symmetry with the constructor and rename (and not make it a `Writable`) after working out how to not write the shard id in that last spot as well. (that's easier to do after we clean up the request side of things in #56092 )

Reply:
Ok